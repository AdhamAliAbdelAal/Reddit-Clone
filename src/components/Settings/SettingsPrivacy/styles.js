import { styled } from '@mui/system';
import { alpha } from '@mui/material/styles';
import {
  TextField,
} from '@mui/material';

import theme, { fonts } from '../../../styles/theme';

export const AddBlock = styled('div')(() => ({
  marginTop: '20px',
  borderRadius: '5px',
  border: '1px solid #EDEFF1',
  display: 'flex',
  height: 52,
  padding: '0 24px',
  position: 'relative',
  '&:focus': {
    border: '1px solid #0079D3',
  },
}));
export const AddButton = styled('div')(({ active }) => ({
  marginLeft: 12,
  fontSize: 14,
  fontWeight: '700',
  color: '#0079D3',
  opacity: (!active) ? '0.4' : '1',
  cursor: (!active) ? 'not-allowed' : 'pointer',
  display: 'flex',
  alignItems: 'center',
}));
export const InputBlock = styled('input')(() => ({
  border: 'none',
  outline: 'none',
  flexGrow: 1,
}));

export const RedditTextField = styled((props) => (
  // eslint-disable-next-line react/jsx-props-no-spreading
  <TextField {...props} size="small" autoComplete="off" />
))(({ clr }) => ({
  '&.MuiTextField-root': {
    width: '100%',
    marginBottom: '10px',
    // input text
    '.MuiInputBase-input': {
      fontSize: '12px',
    },

    // Helper Text
    '.MuiFormHelperText-root': {
      margin: '2px 0px 0px 0px',
      color: theme.palette.error.main,
      fontSize: '12px',
      fontFamily: fonts['system-ui'],
      fontWeight: '600',
    },
  },

  '& .MuiFilledInput-root': {
    backgroundColor: '#fff',
    border: `1px solid ${clr}`,
    overflow: 'hidden',
    '&:hover': {
      backgroundColor: 'transparent',
    },
    '&.Mui-focused': {
      backgroundColor: 'transparent',
      boxShadow: clr ? `${alpha(clr, 0.25)} 0 0 0 2px` : 'none',
      borderColor: `${clr}`,
    },
    '&::after': {
      display: 'none',
    },
    '&::before': {
      display: 'none',
    },
  },

  '.MuiInputLabel-shrink': {
    color: `${theme.palette.neutral.main} !important`,
  },
  '.MuiInputLabel-root': {

    fontWeight: 'bold',
    fontSize: '10px',
    letterSpacing: '0.1em',
  },
}));
export const BlocekConataier = styled('div')(() => ({
  display: 'flex',
  justifyContent: 'space-between',
  padding: '8px',
  marginBottom: '8px',
}));

export const ImageBlock = styled('img')(() => ({
  objectFit: 'cover',
  width: '24px',
  heught: '24px',
}));
export const BlocekInfo = styled('div')(() => ({
  display: 'flex',
  gap: '8px',
  padding: '6px 10px',
  cursor: 'pointer',

  '&:hover': {
    backgroundColor: '#EDEFF1',
  },
}));
