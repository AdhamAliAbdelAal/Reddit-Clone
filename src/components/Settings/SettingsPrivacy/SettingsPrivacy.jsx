import { Box } from '@mui/system';
import {
  Header, Content, ContentHeader,
  SettingsPageConranier, SubHeader, ContentSubHeader, Text,
} from '../styles';
import { AddBlock, AddButton, RedditTextField } from './styles';
/**
 * - SettingsPrivacy
 * - Edit block list in Settings page
 *  @component
 */
function SettingsPrivacy() {
  return (
    <SettingsPageConranier>
      <Box sx={{ maxWidth: '688px', flex: '1 1 auto' }}>
        <Header>Safety & Privacy</Header>
        <ContentSubHeader sx={{ marginBottom: '32px' }}>Manage how we use data to personalize your Reddit experience, and control how other redditors interact with you.</ContentSubHeader>
        <SubHeader>SAFETY</SubHeader>
        <Text>
          <Content>
            <ContentHeader>
              People You’ve Blocked
            </ContentHeader>
            <ContentSubHeader>
              Blocked people can’t send you chat requests or private messages.
            </ContentSubHeader>
          </Content>
          <AddBlock>
            <RedditTextField
              label="BLOCK NEW USER"
              variant="filled"
              type="text"
            />
            <AddButton>ADD</AddButton>
          </AddBlock>
        </Text>
      </Box>
    </SettingsPageConranier>
  );
}

export default SettingsPrivacy;
