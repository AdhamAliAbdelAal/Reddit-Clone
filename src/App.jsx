import { BrowserRouter as Router, Routes, Route } from 'react-router-dom';
import { CssBaseline, ThemeProvider } from '@mui/material';

import MainNavBar from './components/MainNavBar/MainNavBar';
import SettingsProfile from './components/Settings/SettingsProfile/SettingsProfile';
import SettingsPrivacy from './components/Settings/SettingsPrivacy/SettingsPrivacy';
import SettingsFeed from './components/Settings/SettingsFeed/SettingsFeed';
import SettingsAccount from './components/Settings/SettingsAccount/SettingsAccount';
import Settings from './pages/Settings';

import SignUpPage from './pages/SignUpPage';
import LogInPage from './pages/LogInPage';
import ForgetUsernamePage from './pages/ForgetUsernamePage';
import ForgetPasswordPage from './pages/ForgetPasswordPage';
import ResetPasswordPage from './pages/ResetPasswordPage';

import HomePage from './pages/HomePage';
import Messages from './pages/Messages';
import Moderation from './pages/Moderation';
import Profile from './pages/Profile';
import Search from './pages/Search';
import theme from './styles/theme';
import CreatePost from './pages/CreatePost';
import PostTypeContextProvider from './contexts/PostTypeContext';
import Notifications from './pages/Notifications';
import SubReddit from './pages/SubReddit';

function App() {
  return (
    <ThemeProvider theme={theme}>
      <CssBaseline />
      <PostTypeContextProvider>
        <Router className="App">
          <MainNavBar />
          <Routes>
            <Route
              path="/"
              element={
                <HomePage data-testid="home page" />
          }
            />
            <Route
              path="/:postClass"
              element={
                <HomePage />
          }
            />
            <Route
              path="/user"
              element={
                <Profile />
              }
            />
            <Route
              path="/register"
              element={
                <SignUpPage />
}
            />
            <Route
              path="/user/:username/"
              element={
                <Profile />
          }
            >
              <Route
                path=":subTitle"
                element={
                  <Profile />
          }
              >
                <Route
                  path=":sort"
                  element={
                    <Profile />
          }
                />

              </Route>
            </Route>

            <Route
              path="/login"
              element={
                <LogInPage />
          }
            />
            <Route
              path="/password"
              element={
                <ForgetPasswordPage />
          }
            />
            <Route
              path="/resetpassword/:token"
              element={
                <ResetPasswordPage />
          }
            />

            <Route
              path="/username"
              element={
                <ForgetUsernamePage />
          }
            />
            <Route
              path="/search"
              element={
                <Search />
          }
            />
            <Route
              path="/messages"
              element={
                <Messages />
          }
            />
            <Route
              path="/moderation"
              element={
                <Moderation />
          }
            />
            <Route
              path="/submit"
              element={
                <CreatePost />
            }
            />
            <Route
              path="/submit/:subredditName"
              element={
                <CreatePost />
            }
            />
            <Route
              path="/notifications"
              element={
                <Notifications />
          }
            />
            <Route
              path="/SubReddit/:Name"
              element={
                <SubReddit />
          }
            >
              <Route
                path=":postClass"
                element={
                  <SubReddit />
          }
              />
            </Route>
            <Route
              path="/settings"
              element={
                <Settings />
          }
            >
              <Route
                path=""
                element={
                  <SettingsAccount />
          }
              />
              <Route
                path="account"
                element={
                  <SettingsAccount />
          }
              />
              <Route
                path="profile"
                element={
                  <SettingsProfile />
          }
              />
              <Route
                path="privacy"
                element={
                  <SettingsPrivacy />
          }
              />
              <Route
                path="feed"
                element={
                  <SettingsFeed />
          }
              />
            </Route>
          </Routes>
        </Router>
      </PostTypeContextProvider>
    </ThemeProvider>
  );
}

export default App;
