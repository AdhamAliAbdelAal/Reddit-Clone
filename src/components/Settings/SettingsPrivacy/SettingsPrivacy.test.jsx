import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import {
  render, screen,
} from '@testing-library/react';
import SettingsPrivacy from './SettingsPrivacy';
import App from '../../../App';

const mockedUsedNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockedUsedNavigate,
}));
const mockAlert = jest.fn();
jest.mock('react-alert', () => ({
  ...jest.requireActual('react-alert'),
  useAlert: () => mockAlert,
}));
// test snapshot
test('test snapshot', async () => {
  const tree = renderer.create(
    <Router>
      <SettingsPrivacy />
    </Router>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
// render settings-privacy page
test('should render settings-privacy page', () => {
  window.history.pushState({}, '', '/settings/privacy');
  render(<App />);
  expect(screen.getByTestId('settings-privacy')).toBeInTheDocument();
});
