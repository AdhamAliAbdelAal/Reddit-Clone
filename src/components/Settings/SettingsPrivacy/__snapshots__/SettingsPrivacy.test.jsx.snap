// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test snapshot 1`] = `
<div
  className="css-nvyjah"
>
  <div
    className="MuiBox-root css-dp89qc"
  >
    <h2
      className="css-1hww4d4"
    >
      Safety & Privacy
    </h2>
    <div
      className="css-135y8re"
    >
      Manage how we use data to personalize your Reddit experience, and control how other redditors interact with you.
    </div>
    <h2
      className="css-1izwtqu"
    >
      SAFETY
    </h2>
    <div
      className="css-1yjvs5a"
    >
      <div
        className="css-m67mx2"
      >
        <div
          className="css-15qfu71"
        >
          People You’ve Blocked
        </div>
        <div
          className="css-1olqyvz"
        >
          Blocked people can’t send you chat requests or private messages.
        </div>
      </div>
      <div
        className="css-8z2fgs"
      >
        <div
          className="MuiFormControl-root MuiTextField-root css-142yrvc-MuiFormControl-root-MuiTextField-root"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-filled MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-filled css-1mog9k0-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink={false}
            htmlFor=":r0:"
            id=":r0:-label"
          >
            BLOCK NEW USER
          </label>
          <div
            className="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall css-19mk8g1-MuiInputBase-root-MuiFilledInput-root"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoComplete="off"
              autoFocus={false}
              className="MuiInputBase-input MuiFilledInput-input MuiInputBase-inputSizeSmall css-c5v1qu-MuiInputBase-input-MuiFilledInput-input"
              disabled={false}
              id=":r0:"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
            />
          </div>
        </div>
        <div
          className="css-t7v2ja"
        >
          ADD
        </div>
      </div>
    </div>
  </div>
</div>
`;
